services/refCompiler.js

```diff
@@ -0,0 +1,154 @@
+var fs = require('fs');
```

Comment:
please confirm: this code is identical to the in-project implemntation that's being ported

also, see comment in `swagger.js`, but I think this file is better done as standard node module, not a BOS service